### PR TITLE
Create view updater

### PR DESCRIPTION
**Pull request by Xodarap**
It annoys me that we need to have weird quoting on postgres tables. This PR creates something which runs after each migration to update a set of views which are case-insensitively-named copies of all our tables. This lets you e.g. `select * from comments` instead of needing to do `select * from "Comments"`.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203746874865946) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts [hunk 1]
```diff
@@ -65,13 +66,31 @@ export async function runMigration(name: string)
...
+  const db = getSqlClient();
+
+  // We can't assume that certain postgres functions exist because we may not have run the appropriate migration
+  // This wraapper runs the function and ignores if it's not defined yet
+  const safeRun = async (fn) => {
+    if(!db) return;
+
+    await db.any(`DO $$
+      BEGIN
+          PERFORM ${fn}();
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/migrationUtils.ts:78]:
```suggestion
        PERFORM ${fn}();
```

File: packages/lesswrong/server/manualMigrations/2023-01-13-createViewUpdater.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        select table_name,
+          string_agg(format('%I as %s', column_name, column_name), ', ') cols
+        from information_schema."columns"
+        inner join information_schema."tables" using (table_name)
+        where "columns".table_schema = 'public'
+          and table_name not in ('migration_log', 'mongo2pg_lock')
+          and table_type = 'BASE TABLE'
+        group by table_name
+      LOOP
+        EXECUTE format(E'drop view if exists %s; create view %s as select %s from %I;',
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/2023-01-13-createViewUpdater.ts:29]: I'm glad we don't have postgres as a first glass business logic [language](https://github.com/centre-for-effective-altruism/parfit).

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts [hunk 3]
```diff
@@ -65,13 +66,31 @@ export async function runMigration(name: string)
     started: new Date(),
   });
   
+  const db = getSqlClient();
+
+  // We can't assume that certain postgres functions exist because we may not have run the appropriate migration
+  // This wraapper runs the function and ignores if it's not defined yet
+  const safeRun = async (fn) => {
+    if(!db) return;
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/migrationUtils.ts:74]: This seems bad. If the db hasn't been defined yet, throw.

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts [hunk 4]
```diff
@@ -65,13 +66,31 @@ export async function runMigration(name: string)
     started: new Date(),
   });
   
+  const db = getSqlClient();
+
+  // We can't assume that certain postgres functions exist because we may not have run the appropriate migration
+  // This wraapper runs the function and ignores if it's not defined yet
+  const safeRun = async (fn) => {
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/migrationUtils.ts:73]: JP's aesthetic taste is to not make this a closure and instead pass in db when you call it. ¯\\\_(ツ)_/¯

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts [hunk 5]
```diff
@@ -65,13 +66,31 @@ export async function runMigration(name: string)
     started: new Date(),
   });
   
+  const db = getSqlClient();
+
+  // We can't assume that certain postgres functions exist because we may not have run the appropriate migration
+  // This wraapper runs the function and ignores if it's not defined yet
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/migrationUtils.ts:72]:
```suggestion
  // This wrapper runs the function and ignores if it's not defined yet
```

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts (hunk 4 above)
Comment (jpaddison3) [reply packages/lesswrong/server/manualMigrations/migrationUtils.ts:73]: Eh, don't bother changing

**Review by jpaddison3 — APPROVED** [review]
One comment I think is important-ish. Other than that, LGTM. I am in fact a fan of your work, and look forward to using this.

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts (hunk 3 above)
Comment (Xodarap) [reply packages/lesswrong/server/manualMigrations/migrationUtils.ts:74]: Won't that break for less wrong? I guess I don't really understand how the migration architecture is different for them

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts [hunk 6]
```diff
@@ -65,13 +66,31 @@ export async function runMigration(name: string)
...
+          PERFORM ${fn}();
+      EXCEPTION WHEN undefined_function THEN
+        -- Ignore if the function hasn't been defined yet; that just means migrations haven't caught up
+      END;
+    $$;`)
+  }
+
+  // TODO: do this atomically in a single transaction
   try {
+    await safeRun(`remove_lowercase_views`) // Remove any views before we change the underlying tables
```
Comment (oetherington) [inline packages/lesswrong/server/manualMigrations/migrationUtils.ts:87]: I could be misunderstanding, but I think this is in the wrong place.

All (or at least, most) changes to the Postgres schema should be done in automatic migrations, not in manual migrations, so this should probably be called somewhere in `packages/lesswrong/server/migrations/meta/umzug.ts`.

I don't think it hurts to have it here as well though.

File: packages/lesswrong/server/manualMigrations/migrationUtils.ts (hunk 3 above)
Comment (jpaddison3) [reply packages/lesswrong/server/manualMigrations/migrationUtils.ts]: (I don't know the right sol'n)

**Pull request merged**